packages/core/src/application.ts

```diff
@@ -130,6 +131,9 @@ export class Application extends Context {
    * @memberof Application
    */
   public async start(): Promise<void> {
+    await this._forEachComponent(c => {
+      if (isLifeCycleAction(c)) return c.start();
```

Comment:
Ultimately we'll switch the `observer` pattern so that components can subscribe to the `start/stop` events. See #1972.

Do we want to wait unless we settle on #1972?

Reply:
I think that even with the observer pattern from #1972, we should still allow components to export lifecycle observers in an indirect way, i.e. letting `mountComponent` to handle the actual registration.

It is not clear to me how #1972 would affect the design of lifecycle events and whether the solution based purely on low-level observer APIs would be enough easy to use.

My preferred approach is to use composition instead of inheritance, allow components to export an array of life-cycle listener/observe classes and get this pull request landed.

Reply:
Later on, when #1972 is implemented, we can look into ways how to leverage the observer pattern as a low-level implementation detail to power "LifeCycleObserver".